libvmi/os/windows/core.c

```diff
@@ -834,10 +834,33 @@ init_from_json_profile_real(vmi_instance_t vmi, reg_t kpcr_register_to_use)
 
     if (kpcr_register_to_use) {
         reg_t kpcr_reg = 0;
-        dbprint(VMI_DEBUG_MISC, "** Trying kpcr_register_to_use to get KPCR.\n");
-        if (VMI_FAILURE == driver_get_vcpureg(vmi, &kpcr_reg, kpcr_register_to_use, 0)) {
-            dbprint(VMI_DEBUG_MISC, "** driver_get_vcpureg(..) failed.\n");
-            goto done;
+
+        if ((vmi->page_mode == VMI_PM_LEGACY || vmi->page_mode == VMI_PM_PAE) && kpcr_register_to_use == FS_BASE) {
```

Comment:
Please split this logic into a separate function.

Reply:
Split logic into two separate methods, `kpcr_get` and `kpcr_get_x86`.